controllers/notesControllers.js

```diff
@@ -1,33 +1,64 @@
-let notes = [];
-let currentNote;
+const Note = require('../validations/noteModel');
 
-const putNote = (req, res) => {
-    currentNote = notes.find((note) => note.id === req.params.id);
-    if (!currentNote) {
-        return res.status(404).json({ error: { message: 'note not found' } });
+const putNote = async (req, res) => {
+    try {
+        const currentId = req.params.id;
+        const currentNote = await Note.findOne({ id: currentId });
+        if (!currentNote) {
+            throw new Error('note is not found');
+        }
+        if (!req.body) {
+            throw new Error('enter body to send');
+        }
+        await Note.findOneAndUpdate({ id: currentId }, req.body);
+        res.json(req.body);
```

Comment:
you need to return an updated note, so make like this :
const note = await Note.findOneAndUpdate({ id: currentId }, req.body);
res.json(note);